src/js/dp/search.js

```diff
@@ -8,138 +8,173 @@ const fetchHtml = async (url) => {
     );
-    
+
     replaceWithIEPollyfill(
-      searchContainer.querySelector(".search__summary__count"), 
+      searchContainer.querySelector(".search__summary__count"),
       dom.querySelector(".search__summary__count")
     );
-    
+
     initPaginationListeners();
-    
+
     // scroll to the top of the page after the content has been refreshed, to indicate a change has occured
-    if(scrollToTop){
-      const searchResultsSection = searchContainer.querySelector('[aria-label="Search results"]')
-      const resultsSectionOffsetFromTop = searchResultsSection.getBoundingClientRect().top + document.documentElement.scrollTop;
-      window.scrollTo(0, resultsSectionOffsetFromTop)
+    if (scrollToTop) {
+      const searchResultsSection = searchContainer.querySelector(
+        '[aria-label="Search results"]'
+      );
+      const resultsSectionOffsetFromTop =
+        searchResultsSection.getBoundingClientRect().top +
+        document.documentElement.scrollTop;
+      window.scrollTo(0, resultsSectionOffsetFromTop);
     }
   }
-}
+};
 
 const switchCheckbox = (paramsArray) => {
   // get current param
-  let strParams = window.location.search; 
+  let strParams = window.location.search;
 
   // build new param
   paramsArray.map((param) => {
-    if(!('isChecked' in param) || !('filterName' in param)) return;
+    if (!("isChecked" in param) || !("filterName" in param)) return;
     if (param.isChecked) {
       strParams += `&filter=${param.filterName}`;
     } else {
-      strParams = strParams.replace(new RegExp(`(\\&|\\?)filter\=${param.filterName}`), "")
+      strParams = strParams.replace(
+        new RegExp(`(\\&|\\?)filter\=${param.filterName}`),
+        ""
+      );
     }
   });
 
   // make the change to the markup
-  switchSearchMarkup(strParams, true)
-}
+  switchSearchMarkup(strParams, true);
+};
 
-// create listeners for checkboxes controlling eachother
-[...searchContainer.querySelectorAll('.ons-checkboxes__items [aria-controls]')].map(topFilter => {
-  const childrenSelector = topFilter.getAttribute('aria-controls');
-  const theChildren = [...searchContainer.querySelectorAll(`#${childrenSelector} [type=checkbox]`)];
-  if(!childrenSelector) return;
-  topFilter.addEventListener("change", async (e) => {
-      const paramsArray = theChildren.map(item => ({isChecked: e.target.checked, filterName: item.value}));
-      theChildren.map(item => item.checked = e.target.checked);
+if (searchContainer) {
+  // create listeners for checkboxes controlling eachother
+  [
+    ...searchContainer.querySelectorAll(
+      ".ons-checkboxes__items [aria-controls]"
+    ),
+  ].map((topFilter) => {
+    const childrenSelector = topFilter.getAttribute("aria-controls");
+    const theChildren = [
+      ...searchContainer.querySelectorAll(
+        `#${childrenSelector} [type=checkbox]`
+      ),
+    ];
+    if (!childrenSelector) return;
+    topFilter.addEventListener("change", async (e) => {
```

Comment:
`async` unnecessary because it doesn't `await` anything or need to implicitly return a promise.

Reply:
As above